src/main/scala/nl/knaw/dans/api/sword2/DepositHandler.scala

```diff
@@ -367,8 +357,8 @@ object DepositHandler {
       }
   }
 
-  private def compareChecksumAgainstReferredBag(file: String, checksum: String, url: String)(implicit id: String, bs: BagStoreBase): Try[Unit] = {
-    val referredFile = getReferredFile(url, bs.baseUrl)
+  private def compareChecksumAgainstReferredBag(file: String, checksum: String, url: String)(implicit id: String, bs: Option[BagStoreSettings]): Try[Unit] = {
+    val referredFile = getReferredFile(url, bs.get.baseUrl)
```

Comment:
How do you know that `bs` is not a `None` here? You don't! So don't use a `get`.